### ISSUE TITLE
Squiz.Arrays.ArrayDeclaration.SingleLineNotAllowed

### ISSUE THREAD

**Issue by evilebottnawi** [squizlabs/PHP_CodeSniffer#697]
Not check this code on error
```php
add_action('admin_init', [1, 2, 3]);
```
but throw error 
```php
$test = [1, 2, 3];
```

**Comment by gsherwood** [comment]
That's because the sniff allows arrays with multiple values when used inside function calls: https://github.com/squizlabs/PHP_CodeSniffer/blob/master/CodeSniffer/Standards/Squiz/Sniffs/Arrays/ArrayDeclarationSniff.php#L150

When not inside a function call, it enforces that the array be defined over multiple lines for easy editing.

Does that make sense?

**Comment by evilebottnawi** [comment]
@gsherwood u think we need options for this feature,
allow(disallow) multiple declaration array inside function class (and function signature) - (4 options).
what do you think about this?

**Comment by gsherwood** [comment]
I think the Squiz array sniff is so specific that I don't really want to add options to it. Unless people actually want arrays defined in that exact way, it's best not to use it

Instead, I'd like to write a more generic array sniff that just enforces basic indenting, or change this sniff to do that. Then it will be more widely useful and provide a solid array standard that more people can use.

**Comment by evilebottnawi** [comment]
@gsherwood it would be very convenient.

**Comment by gsherwood** [comment]
I'm going to close this and ref it in request #629 - which is to break out the Squiz array sniff into multiple useful sniffs.
